File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Protocol/Log/TableLog.vb
Imports VecompSoftware.Helpers.ExtensionMethods
<Serializable()>
Public Class TableLog
    Inherits AuditableDomainObject(Of Guid)

#Region "Properties"
    Public Overridable Property SystemComputer As String
    Public Overridable Property SystemUser As String
    Public Overridable Property TableName As String
    Public Overridable Property EntityId As Integer
    Public Overridable Property EntityUniqueId As Guid
    Public Overridable Property LogType As LogEvent
    Public Overridable Property LogDescription As String
    Public Overridable Property Hash As String
#End Region

#Region "Ctor/init"
    Public Sub New()
        Id = Guid.NewGuid()
    End Sub
#End Region
End Class

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Commons/TableLog.cs
using System;

namespace VecompSoftware.DocSuiteWeb.Entity.Commons
{
    public class TableLog : DSWBaseEntity
    {
        #region [ Constructor ]

        public TableLog() : this(Guid.NewGuid()) { }

        public TableLog(Guid uniqueId)
            : base(uniqueId)
        { }
        #endregion

        #region [ Properties ]

        public int? LoggedEntityId { get; set; }

        public Guid? LoggedEntityUniqueId { get; set; }

        public TableLogEvent LogType { get; set; }

        public string TableName { get; set; }

        public string SystemComputer { get; set; }

        public string LogDescription { get; set; }

        public string Hash { get; set; }

        #endregion

        #region [ Navigation Properties ]

        #endregion
    }
}
